### PR TITLE
chore: refactor crud methods

### PR DESCRIPTION
**Pull request by AnaghHegde**
## Description
As part of transaction support in PG, we are moving from using the jpa methods for database operations. This PR is refactoring the code to use custom repository class for DatasourceStorageRepository from the default CrudRepository.

## Automation

/ok-to-test tags="@tag.ImportExport"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/12406578294>
> Commit: c7efa6354d2cc1982121eeeab84ac15f1d400e9a
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=12406578294&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.ImportExport`
> Spec:
> <hr>Thu, 19 Dec 2024 05:27:22 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
	- Introduced new methods for retrieving and counting `NewAction` entities based on application ID and IDs collection.

- **Bug Fixes**
	- Removed outdated methods from the `NewActionRepositoryCE` interface to streamline functionality.

- **Refactor**
	- Enhanced repository capabilities for managing `NewAction` entities while maintaining backward compatibility.

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/ce/CustomNewActionRepositoryCEImpl.java [hunk 1]
```diff
@@ -488,4 +490,25 @@ public Flux<NewAction> findAllByApplicationIds(List<String> applicationIds, List
...
+    public Flux<NewAction> findByApplicationId(String applicationId) {
+        return queryBuilder()
+                .criteria(Bridge.equal(NewAction.Fields.applicationId, applicationId))
+                .all();
+    }
+
+    @Override
+    public Flux<NewAction> findAllByIdIn(Iterable<String> ids) {
+        return queryBuilder()
+                .criteria(Bridge.in(NewAction.Fields.applicationId, IteratorUtils.toList((Iterator) ids)))
```
Comment (abhvsn) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/ce/CustomNewActionRepositoryCEImpl.java:504]: Should it be `NewAction.Fields.id`?

File: app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/ce/CustomNewActionRepositoryCE.java [hunk 2]
```diff
@@ -80,4 +81,12 @@ Flux<NewAction> findAllPublishedActionsByContextIdAndContextType(
             String contextId, CreatorContextType contextType, AclPermission permission, boolean includeJs);
 
     Flux<NewAction> findAllByApplicationIds(List<String> branchedArtifactIds, List<String> includedFields);
+
+    @Meta(cursorBatchSize = 10000)
+    Flux<NewAction> findByApplicationId(String applicationId);
+
+    @Meta(cursorBatchSize = 10000)
```
Comment (abhvsn) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/ce/CustomNewActionRepositoryCE.java:88]: How will this annotation work with custom repo classes?

**Review by abhvsn — CHANGES_REQUESTED** [review]
Comments added

File: app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/ce/CustomNewActionRepositoryCE.java (hunk 2 above)
Comment (AnaghHegde) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/ce/CustomNewActionRepositoryCE.java:88]: No it does not work. But the number is very high and we have not reached this. I will add a comment for us to implement this in future.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/ce/CustomNewActionRepositoryCE.java [hunk 3]
```diff
@@ -80,4 +80,14 @@ Flux<NewAction> findAllPublishedActionsByContextIdAndContextType(
             String contextId, CreatorContextType contextType, AclPermission permission, boolean includeJs);
 
     Flux<NewAction> findAllByApplicationIds(List<String> branchedArtifactIds, List<String> includedFields);
+
+    // @Meta(cursorBatchSize = 10000)
+    // TODO Implement cursor with batch size
+    Flux<NewAction> findByApplicationId(String applicationId);
+
+    // @Meta(cursorBatchSize = 10000)
+    // TODO Implement cursor with batch size
```
Comment (abhvsn) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/ce/CustomNewActionRepositoryCE.java:89]: Let's use the .all() method to implement the batching, which will make sure all the DB calls by default will be batched.

**Review by abhvsn — DISMISSED** [review]
Added minor comment.

**Pull request merged**